snap/hooks/post-refresh

```diff
@@ -0,0 +1,43 @@
+#!/bin/sh
+set -eu
+
+PREVIOUS_VERSION=$(snapctl get previous-version)
+INITIAL_ALLOWED_USERS_VERSION="0.2.0"
+
+version_less_than() {
+  [ "$(printf '%s\n' "${@}" | sort -V | head -n1)" = "$1" ]
+}
+
+echo_and_log_to_journal() {
+  echo "${@}"
+  logger -t $SNAP_NAME "${@}"
+}
+
+should_transition_to_allowed_users() {
+  # Transition to allowed users if:
+  # - previous-version is not set (that means that the previous version is
+  #   older than 0.2.0, i.e. the version where we introduced setting the
+  #   previous-version in the pre-refresh hook).
+  # - previous-version is set, but it is less than 0.2.0. That should never
+  #   happen, but we check it to give an example how the previous-version
+  #   can be used to transition data from older versions.
+  [ -z "${PREVIOUS_VERSION}" ] || version_less_than "${PREVIOUS_VERSION}" "${INITIAL_ALLOWED_USERS_VERSION}"
+}
+
+transition_to_allowed_users() {
+  echo_and_log_to_journal "Transitioning to allowed users"
+  mkdir -p ${SNAP_DATA}/broker.conf.d
+  cp --update=none ${SNAP}/conf/migrations/pre-${INITIAL_ALLOWED_USERS_VERSION}/broker.conf.d/* ${SNAP_DATA}/broker.conf.d/
```

Comment:
Nice, as per this I think you can just make the logic easier by just copying everything that is under  `${SNAP}/conf/migrations/pre-$....` where the version is something lower than ours no?

I mean, without having  specific `should_transition_to_allowed_users` case I think.

Reply:
I don't see how you can get rid of the checks done in `should_transition_to_allowed_users`. can you provide the command(s) you have in mind to do that?

Reply:
Yeah I was thinking something like that, but let me know if that's not as simple as I thought :)

Reply:
The problem I see with this is that if you refresh with the same version (or even from 0.2 to 0.3) these files would be created.

We don't want this to happen, because it would changing the default behavior (it would be allow `ALL` instead of allow `OWNER`). We only want to to create those files if we are migrating from `<0.2`, because the default behavior has changed and we don't want to introduce breaking changes.

Reply:
I see, fair enough...